dubbo-registry/dubbo-registry-nacos/src/main/java/org/apache/dubbo/registry/nacos/NacosRegistry.java

```diff
@@ -324,32 +323,32 @@ private void filterServiceNames(Set<String> serviceNames, URL url) {
             // (required) segments[1] = serviceInterface
             // (optional) segments[2] = version
             // (optional) segments[3] = group
-            String[] segments = StringUtils.split(serviceName, SERVICE_NAME_SEPARATOR);
+            String[] segments = serviceName.split(SERVICE_NAME_SEPARATOR, -1);
             int length = segments.length;
             if (length != 4) { // must present 4 segments
                 return false;
             }
 
             String category = segments[CATEGORY_INDEX];
-            if (!ArrayUtils.contains(categories, category)) { // no match category
+            if (!categories.contains(category)) { // no match category
                 return false;
             }
 
             String serviceInterface = segments[SERVICE_INTERFACE_INDEX];
             if (!WILDCARD.equals(targetServiceInterface) &&
-                    !StringUtils.equals(targetServiceInterface, serviceInterface)) { // no match service interface
+                    !targetServiceInterface.equals(serviceInterface)) { // no match service interface
                 return false;
             }
 
             String version = segments[SERVICE_VERSION_INDEX];
             if (!WILDCARD.equals(targetVersion) &&
-                    !StringUtils.equals(targetVersion, version)) { // no match service version
+                    !targetVersion.equals(version)) { // no match service version
```

Comment:
Same here.

Reply:
targetVersion cannot be null, but I will use dubbo's StringUtils anyway.